solvebio/test/test_shortcuts.py

```diff
@@ -59,45 +48,40 @@ def _validate_tmpl_fields(self, fields):
             elif f.name == 'aliases':
                 self.assertEqual(f.data_type, 'string')
 
-    @mock.patch('solvebio.resource.Depository.create',
-                side_effect=fake_depo_create)
-    @mock.patch('solvebio.resource.DepositoryVersion.create',
-                side_effect=fake_depo_version_create)
-    @mock.patch('solvebio.resource.Dataset.create',
-                side_effect=fake_dataset_create)
-    @mock.patch('solvebio.resource.DatasetTemplate.create',
-                side_effect=fake_data_tpl_create)
-    def test_create_dataset_upload_template(
-            self,
-            fake_depo_create,
-            fake_depo_version_create,
-            fake_dataset_create,
-            fake_data_tpl_create):
-        dataset_full_name = self.unique_dataset_name()
```

Comment:
Careful with mock ordering - this can be confusing.  The above mocks are used in the incorrect order.  This wasn't an issue because they were unused, but the correct order is reverse.  Example below.

```
import time
import unittest

import mock


class Tests(unittest.TestCase):

    @mock.patch('time.time', side_effect=lambda: 123)
    @mock.patch('time.asctime', side_effect=lambda: 'that asc time')
    def test_mocks(self, mock1, mock2):
        self.assertEqual(1, 1)

        self.assertEqual(time.time(), 123)
        self.assertEqual(time.asctime(), 'that asc time')

        # The outer decorators wrap inner decorators.  Mocks are passed in
        # to the original function such that the closest mock is furthest left.
        self.assertEqual(mock1(), 'that asc time')
        self.assertEqual(mock2(), 123)


# Recall that the decorated function below:

@make_bold
@make_italic
def hello():
  return "hello world"
  
# Is the same as...

def hello():
  return "hello world"
hello = make_bold(make_italic(hello))
```